### PR TITLE
Retired items#41

### PR DESCRIPTION
**Pull request by CjMoore**
- adds column for item_status
- adds enum for item_status
- if item is retired user cannot add to cart in show or index
- cleans up items model to user shoulda matchers
- adds model test for item_status enum
- fixes fabricator

### CODE REVIEW

File: spec/models/item_spec.rb [hunk 1]
```diff
@@ -1,34 +1,19 @@
...
-        item = Item.new(title: "Bang Bang Boggart Banger",  price: 10.99 )
-
-        expect(item).to be_invalid
-      end
+  context "validations" do
+    it { is_expected.to validate_presence_of(:title) }
+    it { is_expected.to validate_uniqueness_of(:title) }
+    it { is_expected.to validate_presence_of(:description) }
+    it { is_expected.to validate_uniqueness_of(:description) }
+    it { is_expected.to validate_presence_of(:price) }
```
Comment (meyerhoferc) [inline spec/models/item_spec.rb:9]: 👍 for shoulda

File: app/models/item.rb [hunk 2]
```diff
@@ -1,5 +1,7 @@
 class Item < ApplicationRecord
   validates :title, :description, :price, presence: true
   validates :title, :description, uniqueness: true
-  belongs_to :category 
+  belongs_to :category
+
+  enum item_status: ['available', 'retired']
```
Comment (meyerhoferc) [inline app/models/item.rb:6]: looks great

File: app/views/shared/_items.html.erb [hunk 3]
```diff
@@ -10,7 +10,11 @@
           <p>
             <%= item.price %>
           </p>
-          <%= button_to "Add to Cart", carts_path(item_id: item.id) %>
+          <% if item.available? %>
+            <%= button_to "Add to Cart", carts_path(item_id: item.id) %>
+          <% else %>
+            <%= button_to "Retired", class: "btn disabled" %>
+          <% end %>
```
Comment (meyerhoferc) [inline app/views/shared/_items.html.erb:17]: 🔥

File: db/migrate/20170223203005_add_item_status_to_items.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+class AddItemStatusToItems < ActiveRecord::Migration[5.0]
+  def change
+    add_column :items, :item_status, :integer, default: 0
```
Comment (meyerhoferc) [inline db/migrate/20170223203005_add_item_status_to_items.rb:3]: looks like the default is available, great

File: spec/fabricators/fabricator.rb [hunk 5]
```diff
@@ -4,7 +4,7 @@
   title { Faker::HarryPotter.book }
   description { Faker::HarryPotter.quote }
   price { Faker:: Number.decimal(2) }
-  category  
+  category  { Fabricate(:category) }
```
Comment (meyerhoferc) [inline spec/fabricators/fabricator.rb:7]: so glad this is working

**Review by meyerhoferc — APPROVED** [review]
This looks great. There's a lot of things fixed in here! Ready for @valerietrudell or @cmacaulay to merge

File: spec/fabricators/fabricator.rb (hunk 5 above)
Comment (valerietrudell) [inline spec/fabricators/fabricator.rb]: sweet; good work on fabricator lets use it for the parts that we have in our tests

**Pull request merged**